Task 2 - all elements of learning outcome minus Page Factory

William-436 opened the pull request:
Task 2 Learning Outcomes: spreadsheets, POM, data-driven approach, NUnit, Extent Reports, Excel Data Reader, AutoIt, and Asserts

William-436 committed:
spreadsheets and all code for Task2 covering POM, data-driven, NUnit, Extent Reports, Excel Data Reader, AutoIt, and Asserts

amanbansalnz commented on MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/HomePage.cs:
```diff
@@ -0,0 +1,100 @@
+        private IWebElement signinLink => driver.FindElement(By.XPath(signinLinkXP));
+
+        // find the login button
+        //[FindsBy(How = How.XPath, Using = "/html/body/div[2]/div/div/div[1]/div/div[4]/button")]
```
remember to remove all unwanted comments

William-436 replied:
Removed unwanted comments from HomePage.cs

amanbansalnz commented on MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ManageListingsPage.cs:
```diff
@@ -0,0 +1,234 @@
+
+        // this method is only called by positive test cases containing valid data
+        public bool CompareManageListingsValues(int NumberOfExistingListingsInt, string Category, string Title, string Description, string ServiceType, string SkillTrade, string Active)
+        {
```
can you please break this method into smaller methods, this way your code is more readable and easy to understand

William-436 replied:
Removed 1 parameter from parms and created separate methods to check each parameter

amanbansalnz commented on MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs:
```diff
@@ -0,0 +1,1144 @@
+        //public static string capturedMessage;
+
+    
+        public void CreateShareSkill(string ActionOfTest, string ValidInvalid, string ActionButton, string FollowupAction, string Title, string TitleError, string Description, string DescriptionError, string Category, string Subcategory, string Tags, string ServiceType, string Location, string StartDate, string EndDate, string StartEndDateError, string Sun, string SunStartTime, string SunEndTime, ...
```
so this should be broken down into smaller methods! Also the number of args you are passing in this method is way too much think about other means of doing this! either exposing methods which take less number of arguments or introduce a new object that sets all these fields and pass that object to this method

William-436 replied:
Replaced CreateShareSkill method with separated method for each input field

amanbansalnz commented on MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs:
```diff
@@ -0,0 +1,1144 @@
+        }
+
+        public string ConvertDate(string DatetoConvert)
+        {
```
same with this method, break down into smaller methods also remove comments as its hard to read the code

William-436 replied:
Modified ConvertDate method to use a new method (CalculateTheDate) to execute reusable lines of code. Also cleaned up excessive comments and left informative comments.

amanbansalnz commented on MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs:
```diff
@@ -0,0 +1,181 @@
+        {
+            Console.WriteLine("**Starting 'Trade Your Skills' portal script from CommonDriver");
+
+            // create the Extent Report
```
again break down into readable methods

William-436 replied:
Please clarify reason for breaking down the SetupActions method of the CommonDriver's non-repeatable code into separate methods. I have attempted to explain each portion of code in the SetupActions method with a clear comment. Are the comments unclear or should the portions of code be grouped in a different sequence?

William-436 replied:
Resolving conversation as mentor approved submission without requiring this update to be completed.

amanbansalnz commented on MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs:
```diff
@@ -0,0 +1,181 @@
+            {
+                //if (homePageObj.SignOut() == true)
+                //{
+                //    Console.WriteLine("Successfully Signed out of 'Trade Your Skills' portal");
```
remove comments

William-436 replied:
Removed unnecessary comments from CommonDriver.cs

William-436 commented:
The 'Conflicting file' is Task 1's deliverable which I am awaiting feedback on its review. If you approve it without changes, then I am happy to simply remove it from the repository and upload an updated version of the same file containing some corrections and enhancements. Please advise on the status of its review.

William-436 committed:
Updates per feedback in PR

William-436 commented:
Deleted spreadsheet containing test conditions and test cases from the main branch to resolve merge conflict since newer, updated spreadsheet in the branch is intended to replace the spreadsheet in the main branch.
